extensions/Worker.Extensions.Storage.Queues/src/QueueMessageJsonConverter.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Storage.Queues.Models;
+
+namespace Microsoft.Azure.Functions.Worker.Storage.Queues
+{
+    internal class QueueMessageJsonConverter : JsonConverter<QueueMessage>
+    {
+        public override bool CanConvert(Type objectType) => objectType == typeof(QueueMessage);
+
+        public override QueueMessage? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType is not JsonTokenType.StartObject)
+            {
+                throw new JsonException("JSON payload expected to start with StartObject token.");
+            }
+
+            string messageId = String.Empty;
+            string popReceipt = String.Empty;
+            string messageText = String.Empty;
+            long dequeueCount = 1;
+            DateTime? nextVisibleOn = null;
+            DateTime? insertedOn = null;
+            DateTime? expiresOn = null;
+
+            var startDepth = reader.CurrentDepth;
+
+            while (reader.Read())
+            {
+                if (reader.TokenType is JsonTokenType.EndObject && reader.CurrentDepth == startDepth)
+                {
+                    return QueuesModelFactory.QueueMessage(
+                        messageId,
+                        popReceipt,
+                        messageText,
+                        dequeueCount,
+                        nextVisibleOn,
+                        insertedOn,
+                        expiresOn
+                    );
+                }
+
+                if (reader.TokenType is not JsonTokenType.PropertyName)
+                {
+                    continue;
+                }
+
+                var propertyName = reader.GetString();
+                reader.Read();
+
+                switch (propertyName)
+                {
+                    // We're not expecting these three values to be null. If they are, should we throw or just return an empty string?
+                    case "MessageId":
```

Comment:
Do we need case insensitive support?

Reply:
I'm not 100% sure. My initial though is no, this should always be CamelCase - but I don't think it hurts to account for insensitive casing?